java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaObjectOutputStream.java

```diff
@@ -170,54 +177,63 @@ public Thread newThread(final Runnable runnable) {
          * We have to wait here until the upload to Manta starts and a Writer
          * becomes available.
          */
-        while (httpContent.getWriter() == null) {
-            try {
-                Thread.sleep(CLOSED_CHECK_INTERVAL);
-            } catch (InterruptedException e) {
-                return;
+        synchronized (this.httpContent) {
```

Comment:
If I have github lined up right, this synchronized block is inside the constructor.  Is this field even published yet in a way another thread could content on?

Reply:
Oops: https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait--

"The current thread must own this object's monitor. "